### ISSUE TITLE
Tabs as webcomponent

### ISSUE THREAD

**Issue by ffriedl89** [dynatrace-oss/barista#1178]
# Feature Request

![image](https://user-images.githubusercontent.com/2905693/85258159-432ee780-b467-11ea-9b7f-bf690f7871f8.png)

We need tabs for the Next project. For the first version create a grouping component and a single tab component as a web component.
Leverage design tokens for styling. 

Read up on a11y before starting 
Check a11y information like https://inclusive-components.design/tabbed-interfaces/
https://www.w3.org/TR/wai-aria-practices-1.1/#tabpanel

No lazy loading - load all tabs eagerly for first version - lazy loading (creating the tab content once the tab gets active is not part of this issue) 

Sync with UX on design specifics. 

Create an API proposal outlining the first version of this component

**Comment by rowa-audil** [comment]
# API Proposal for the __Tabs and Tab webcomponent__

## Overview

The **Tabs** web component wrappes **Tab** web components based on an array and will dispatch an event or set the currently active section after clicking on one of the Tabs.

## API
### Tab
#### Input:
- sectionId: string,
#### Output:
- sectionId: EventEmitter<string> and let the wrapper handle which section to show
- or setting the active section in the tab component via jQuery (might actually be the better choice)

### Tabs
none. Behaves like a grouping component

### HTML prototype

```ts
const tabs: string[] = ['section1', 'section2', 'section3']
```
```html
<fluid-tabs>
  <fluid-tab *ngFor="let tab of tabs" [sectionId]="tab.id"></fluid-tab>
  <!--Content goes here. Contained in section elements-->
  <section #section1>
    <h4>Section 1</h4>
  </section>
  <section #section2>
    <h4>Section 2</h4>
  </section>
  <section #section3>
    <h4>Section 3</h4>
  </section>
</fluid-tabs>
```

**Comment by tomheller** [comment]
- For now, we want to move the content that is being shown out of the `fluid-tabs`, so please exclude it from there. 
- Please name the wrapper `fluid-tab-group`.
- The fluid tab group should have an event being fired when a tab is selected
- Please figure out and describe how a tab is set active (is that set on the group or on the actual tab)?
- Tabs should have id / unique identifier property.

Please outline all properties on the tab-group and on the tab itself with their types and what they do in more detail.

**Comment by rowa-audil** [comment]
# Updated API Proposal
## Tab
### Properties
```ts
tabId: string // Contains the id reference to the section/content that needs to be displayed
disabled: boolean // Whether the Tab is disabled or not
```

### Events
```ts
@change changeTabActiveEvent(activeTabId: string) // Fires when a tab is activated by clicking or on keydown. Then propagates that event to the tab group
```
The event will support the keyboard as well. Currently, I'm basing the support on the arrow keys. Pressing the tabulator should set the focus to focusable elements inside the content of the tab group.

### Methods
A method that sets the clicked Tab to active for the tab-group to listen to that event

## Tab Group
### Properties
```ts
disabled: Boolean // Sets all tabs to disabled or not
```

### Events
```ts
@change dispatchChangeActiveTabEvent(activeTabId: string)
```

**Comment by tomheller** [comment]
This looks much better. I think we can remove the `disabled` on the tabgroup. I don't really see a benefit of disabling a whole tab group.

Maybe the tab-group will need an input that let's the consumer set the active tab right away. 

```
// Which tab is set active.
// Should default to the first tab available, if none is set from the outside.
@property active: string;
```

This way, a consumer could define which tab should be active initially.
